crates/typst-syntax/src/parser.rs

```diff
@@ -1824,16 +1766,28 @@ impl<'s> Parser<'s> {
         let at = self.at(kind);
         if at {
             self.eat();
-        } else if kind == SyntaxKind::Ident && self.current.is_keyword() {
+        } else if kind == SyntaxKind::Ident && self.current.kind.is_keyword() {
             self.trim_errors();
-            self.eat_and_get().expected(kind.name());
+            self.eat_as_expecting(Some(kind.name()));
         } else {
             self.balanced &= !kind.is_grouping();
             self.expected(kind.name());
         }
         at
     }
 
+    /// Eat the current node and convert to expected or unexpected for Some/None expected.
+    #[track_caller]
+    fn eat_as_expecting(&mut self, expected: Option<&str>) {
```

Comment:
It doesn't really feel worth it to me to take `Option` here and make every caller wrap in `Some(_)` just to not copy three lines into `unexpected`.

Reply:
I realized that this can actually be even further simplified now that we have access to the node itself by just calling `current.node.unexpected()` or `current.node.expected(thing)` *before* calling `p.eat()`. I've removed the function and pulled those out to the three callers in the new "small changes" commit.